lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -282,12 +283,20 @@ pub(crate) fn create_payload_storage(
     Ok(payload_storage)
 }
 
-pub(crate) fn create_id_tracker(database: Arc<RwLock<DB>>) -> OperationResult<IdTrackerEnum> {
+pub(crate) fn create_mutable_id_tracker(
+    database: Arc<RwLock<DB>>,
+) -> OperationResult<IdTrackerEnum> {
```

Comment:
Would be more convenient, if this function returns just `SimpleIdTracker`, not enum